### PR TITLE
Rework node C++  related glossary updates

### PR DESCRIPTION
**Pull request by springmeyer**
This overhauls the glossary items related to Node C++. Specifically it:

 - Fixes missing definitions
 - Points to node-cpp.md from more glossary items
 - Removes content from the glossary that is already in node-cpp.md
 - Moves the `I/O` entry into the general terms (since it is not node specific)
 - Simplifies and clarifies the definitions.

/cc @GretaCB for review (please merge when you feel this is ready).

/cc @nickidlugash for visibility (was pairing with @nickidlugash when I noticed the missing glossary items)

### CODE REVIEW

File: glossary.md [hunk 1]
```diff
@@ -1436,51 +1442,46 @@ Often static is used in combination with [constexpr](#constexpr) to ask for a va
...
--   Threadpool: To do [file I/O](https://github.com/mapbox/cpp/blob/master/glossary.md#io) or any other blocking system call
--   Event Loop: A main process (or main thread) that responds to an event queue: grabs the top item in the event queue, executes it, and then grabs the next item.
--   Threadpool and Event Loop Manager ([libuv](http://libuv.org/)): This library helps manage the threadpool and event loop, and helps handle asynchronous [I/O operations](https://github.com/mapbox/cpp/blob/master/glossary.md#io).
+It is a command line tool that can execute Javascript files and C++ code compiled as [a node addon](#node-addon).
 
-See these articles to understand more about node [performance](#performance):
+Node builds on top of [V8](#v8) by providing ways to do evented, non-blocking [I/O](#io). What this means in practice is that node runs your code on an [event loop](#event-loop) and provides an API for deferring work through asynchronous function callbacks which execute in a [threadpool](#node-threadpool). This "work" might be traditional [I/O](#io) like accessing the filesystem or a data on a network, or it could be any kind of intensive computations.
```
Comment (GretaCB) [inline glossary.md]: `#node-threadpool` --> `#threadpool`

File: glossary.md [hunk 2]
```diff
@@ -1436,51 +1442,46 @@ Often static is used in combination with [constexpr](#constexpr) to ask for a va
...
+In Node.js this is a queue of threads - of fixed size - that are launched at started and ready to execute work.
 
-### threadpool
+To learn how to control the threadpool size with `UV_THREADPOOL_SIZE` and much more see the [node addon docs](https://github.com/mapbox/cpp/blob/master/node-cpp.md)
 
-### worker
+### libuv
 
-### C++ bindings
+A library that provides a cross-platform API for asynchronous [I/O](#io). Most importantly libuv provides access, for [node addons](#node-addon), to the [Node Threadpool](#node-threadpool).
```
Comment (GretaCB) [inline glossary.md]: `#node-threadpool` --> `#threadpool`

File: glossary.md [hunk 3]
```diff
@@ -1436,51 +1442,46 @@ Often static is used in combination with [constexpr](#constexpr) to ask for a va
...
+
+V8 is [open source and written in C++](https://github.com/v8/v8).
 
 ### Node Addon
 
-See [node addon docs](https://github.com/mapbox/cpp/blob/master/node-cpp.md)
+Node provides some "addons" built in, like the `zlib` module, which is implemented [in C++ inside node core](https://github.com/nodejs/node/blob/df63e534584a54dcf02b37446e1e821382e3cef3/src/node_zlib.cc).
 
-### event loop
+You can also write your own [a node addon](#node-addon) and include it dynamically within your application. If you write a [node addon](#node-addon) you can get access to all lower-level operations available in C++ like like working with threads and custom memory allocation.
```
Comment (GretaCB) [inline glossary.md]: Should the node addon links point to Node-Addon's own glossary term here? Or the node-addon.md?

File: glossary.md (hunk 2 above)
Comment (springmeyer) [reply glossary.md]: Thanks, fixed in d29c8d6

File: glossary.md (hunk 1 above)
Comment (springmeyer) [reply glossary.md]: Thanks, fixed in d29c8d6

File: glossary.md (hunk 3 above)
Comment (springmeyer) [reply glossary.md]: Good point. Fixed it by only linking to the `.md` for the text that is within the definition.

**Comment by GretaCB** [comment]
Thank you @springmeyer ! 👍

**Pull request merged**